compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -115,12 +115,7 @@ pub struct InferCtxtInner<'tcx> {
     float_unification_storage: ut::UnificationTableStorage<ty::FloatVid>,
 
     /// Tracks the set of region variables and the constraints between them.
-    ///
-    /// This is initially `Some(_)` but when
-    /// `resolve_regions_and_report_errors` is invoked, this gets set to `None`
-    /// -- further attempts to perform unification, etc., may fail if new
-    /// region constraints would've been added.
-    region_constraint_storage: Option<RegionConstraintStorage<'tcx>>,
```

Comment:
I think it's worth keeping this `Option` and  clearing it in InferCtxt::{resolve_regions,skip_region_resolution} . This way limit this hack to MIR borrowck, rather that lexical region resolver which is more commonly used.